### ISSUE TITLE
The `FormatString` and `ToString` protoflux nodes have different cultural formatting behavior

### ISSUE THREAD

**Issue by Banane9** [Yellow-Dog-Man/Resonite-Issues#429]
### Describe the bug?

This was prompted by #207.

The "FormatString" node aka just "Format" defaults to the user's OS's culture (de-DE for me).
The "ToString" node on the other hand defaults to InvariantCulture (aka US-defaultism central).

### To Reproduce

* Use a non-US culture OS.
* Pick a datatype that gets formatted differently, e.g. floats or dates
* Grab ToString and Format (or Format DateTime as Date) nodes
* Observe that the outputs are different

### Expected behavior

Coming from C#, the expected behavior is to use the OS's culture, though within the context of the game, I'd argue it should be using the display language's culture. Alternatively, both defaulting to InvariantCulture to prevent issues with string formatting for json and such would also be good. People love to build systems using Format to construct json payloads, without knowing / considering, that it could fail on other locales.

### Screenshots

![Inconsistent behavior with float](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/2124570/d2d166ca-797b-4e94-aa36-fc8a1b1ed459)
Inconsistent behavior in the formatting of a float

### Resonite Version Number

2023.10.20.831

### What Platforms does this occur on?

Windows

### Log Files

This doesn't get logged.

### Reporters

Banane9

**Comment by ProbablePrime** [comment]
Yeah, will take a look when i can. Thanks

**Comment by ThomFox** [comment]
This is actually behaving the same as the legacy it inherited. It used to be like this before.

**Comment by Banane9** [comment]
> This is actually behaving the same as the legacy it inherited. It used to be like this before.

Yes, I had an issue about that too: https://github.com/Neos-Metaverse/NeosPublic/issues/2492
